snowpack/src/sources/local.ts

```diff
@@ -624,20 +622,24 @@ export class PackageSourceLocal implements PackageSource {
       const dependencyFileLoc = path.join(installDest, importMap.imports[spec]);
       const loadedFile = await fs.readFile(dependencyFileLoc!);
       if (isJavaScript(dependencyFileLoc)) {
-        const packageImports = new Set<string>();
+        const newPackageImports = new Set<string>();
         const code = loadedFile.toString('utf8');
         for (const imp of await scanCodeImportsExports(code)) {
-          const spec = code.substring(imp.s, imp.e).replace(/(\/|\\)+$/, ''); // remove trailing slash from end of specifier (easier for Node to resolve)
-          if (isRemoteUrl(spec)) {
-            continue;
+          const scannedImport = code.substring(imp.s, imp.e).replace(/(\/|\\)+$/, ''); // remove trailing slash from end of specifier (easier for Node to resolve)
+          if (isRemoteUrl(scannedImport)) {
+            continue; // ignore remote files
           }
-          if (isPathImport(spec)) {
+          if (isPathImport(scannedImport)) {
             continue;
           }
-          packageImports.add(spec);
+          const [scannedPackageName] = parsePackageImportSpecifier(scannedImport);
+          if (scannedPackageName && memoizedImportMap[scannedPackageName]) {
```

Comment:
This is the actual fix here: checking to see if we’ve already installed this package (import map means install happened successfully). Otherwise, we would end up stuck in an infinite loop installing circular npm deps.